src/main/java/service/Preprocessor.java

```diff
@@ -0,0 +1,46 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Preprocessor {
+
+    private final Pattern compile = Pattern.compile("\\d+|[+\\-*/]");
+
+    public List<String> process(String calculationFormula) {
+
+        List<String> values = new ArrayList<>();
+
+        boolean isVerification = verificationOfInputValue(calculationFormula);
+        if(!isVerification) {
+            return values;
```

Comment:
```suggestion
            return Collections.emptyList();
```
비어있는 리스트를 반환할 때 `emptyList` 사용을 권장드립니다. 왜 그럴까요?

Reply:
아마 new ArrayList()를 생성하면 계속해서 빈 객체를 생성해서 할당하는데 emptyList를 사용하면 계속해서 생성하지 않고 동일한 인스턴스를 반환해서 메모리 성능의 효율성도 높이고 가독성도 좋아져서일까요?